README.md

```diff
@@ -31,6 +31,124 @@ A [tutorial](https://github.com/bids-standard/bids-matlab/blob/master/examples/t
+        beh: [0×0 struct]
+        dwi: [0×0 struct]
+        eeg: [0×0 struct]
+        meg: [0×0 struct]
+       ieeg: [0×0 struct]
+        pet: [0×0 struct]
+```
+
+### User interface
+
+Users are not meant to directly interface with the BIDS layout, but rather use the `query` function. In general that works like
+
+```
+BIDS.query(layout, 'item') returns a complete list of the specific items
+BIDS.query(layout, 'item', ...) allows additional key-val pairs for selection
+```
+
+#### Querying for arrays of strings
+
+```
+BIDS.query(layout, 'subjects') returns an array of strings like {'01', '02', ...}
+BIDS.query(layout, 'sessions') returns an array of strings
+BIDS.query(layout, 'modalities') returns an array of strings, like {'eeg', 'anat'}
+BIDS.query(layout, 'types') returns an array of strings, like {'eeg', 'T1w', 'T2w'}
+BIDS.query(layout, 'tasks') returns an array of strings
+BIDS.query(layout, 'runs') returns an array of strings
+```
+
+For subjects, sessions and modalities it can come from the directory structure. For tasks and runs it has to come from the `key-value` entities in the file names. The same could also be used for subjects ('sub-xxx') and sessions ('ses-xxx'), which are in the file name, but the modalities are not as key-value pairs in the file name.
+
+I propose that when you specify
+
+```
+BIDS.query(layout, 'subjects') returns an array of strings like {'01', '02', ...}
+```
+
+it comes from the directory names, and if you specify
+
+```
+BIDS.query(layout, 'subs') returns an array of strings like {'01', '02', ...}
+```
+
+it comes from the 'key-value' entities in the file names. This can be generalised such that `acqs`, `procs`, `dirs` and any [entity](https://bids-specification.readthedocs.io/en/stable/99-appendices/04-entity-table.html) appended with an `s` can be processed. A general implementation also supports future or custom entities (e.g. from non-merged BEPs).
+
+That design choice result in the general
+
+```
+BIDS.query(layout, entity) returns an array of strings
+```
+
+where entity is a string with an `s` appended that matches one of the key-value pairs in the file names. The subjects, sessions, modalities and types remain as explicit queries.
+
+#### Querying for data
+
+The following returns a list of _data_ file names. This excludes the files that come along with the data files, such as the electrodes, channels, photos. For multi-file formats, like BrainVision, it returns only one file name per dataset. For directory based formats, like CTF, it returns the directory name.
+
+```
+BIDS.query(layout, 'data') returns an array of strings
+```
+
+All other queries of metadata should return an array that is the same size as the array of data files.
+
+#### Querying for different types of metadata
+
+The following returns the content of the JSON files that correspond to the data files (see above). In principle it could also return a list of strings with the file names of the JSON files.
```

Comment:
whether to return an array of file names or an array with actual content is discussed in #16.